src/internal.c

```diff
@@ -14758,32 +14758,40 @@ static int DoHandShakeMsgType(WOLFSSL* ssl, byte* input, word32* inOutIdx,
         }
     }
 
-#ifdef OPENSSL_EXTRA
-    if (ssl->CBIS != NULL){
-        ssl->cbmode = SSL_CB_MODE_READ;
-        ssl->cbtype = type;
-        ssl->CBIS(ssl, SSL_CB_ACCEPT_LOOP, SSL_SUCCESS);
-    }
-#endif
-
     if (ssl->options.side == WOLFSSL_CLIENT_END) {
         switch (type) {
         case certificate:
         case server_key_exchange:
         case certificate_request:
         case server_hello_done:
             if (ssl->options.resuming) {
+#ifdef WOLFSSL_WPAS
                 /* This can occur when ssl->sessionSecretCb is set. EAP-FAST
                  * (RFC 4851) allows for detecting server session resumption
                  * based on the msg received after the ServerHello. */
                 WOLFSSL_MSG("Not resuming as thought");
                 ssl->options.resuming = 0;
                 /* No longer resuming, reset peer authentication state. */
                 ssl->options.peerAuthGood = 0;
+#else
+                /* Fatal error. Only try to send an alert. RFC 5246 does not
```

Comment:
so we set `ssl->options.resuming` to 1 only when doing resumption now? Are we sure that we don't error out on a legitim situation?

Reply:
If we set `resuming` then we expect a resumption handshake (session id, ticket, psk). Otherwise we should error out here since the peer is not consistent with their intention. 

Previously we would proceed with the handshake and mark it as unauthenticated in the end. This would result either in a decrypt error or handshake failure because we don't do the necessary key derivation and exchange.